odxtools/diagservice.py

```diff
@@ -187,7 +187,12 @@ def _resolve_snrefs(self, context: SnRefContext) -> None:
         # here because ComparamInstance.short_name is only valid after
         # reference resolution
         self._comparams = NamedItemList(self.comparam_refs)
-
+        if self._request:
+            self._request._resolve_snrefs(context)
+        for response in self._positive_responses:
+            response._resolve_snrefs(context)
+        for response in self._negative_responses:
+            response._resolve_snrefs(context)
```

Comment:
while this might fix the specific problem which you've encountered, this is pretty much a coincidence: this causes the referenced objects to be retargeted to basically the last ECU variant encountered during the reference resolution procedure. If you're looking at another diagnostic layer, e.g. the base variant, the snrefs will be incorrect...

Reply:
yes, u are correct. So with this change, I need to call _resolve_snrefs again before I try to visit other variants. Since the child variant would use parent objects (References) directly, I cannot find other solutions for this problem.

Reply:
I guess #374 might make your life considerably easier?!